### PR TITLE
feat(chart): added the ability to specify K8s extra objects

### LINKED ISSUE #5633: [FR] Allow the Cluster Autoscaler chart to create extra manifest
**Which component are you using?**:

`cluster-autoscaler-chart`

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

Sometimes, accessories to the installation may be manifests to be installed (for example, a SecretProviderClass or ConfigMap).

**Describe the solution you'd like.**:

It would be nice to be able to do this directly via the Chart using the approach that [Grafana](https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L1191), [Prometheus](https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L1064) or [OAuth2 Proxy](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/values.yaml#L333) also uses.

**Describe any alternative solutions you've considered.**:

I don't think there are alternatives

**Additional context.**:

No additional context

### PR DESCRIPTION
**Pull request by pierluigilenoci**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add the ability to specify K8s extra objects

#### Which issue(s) this PR fixes:

Fixes #5633

#### Does this PR introduce a user-facing change?

```release-note
Added the ability to specify K8s extra objects.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@gjtempleton, could you please take a look?

**Comment by pierluigilenoci** [comment]
@mwielgus, please take a look.

File: charts/cluster-autoscaler/Chart.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ name: cluster-autoscaler
 sources:
   - https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
 type: application
-version: 9.37.0
+version: 9.37.1
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/Chart.yaml:14]: We'll need to bump this again as we merged a PR earlier which took us to 9.38.

I'd also want this to be a minor version bump given we're adding functionality.

**Comment by gjtempleton** [comment]
Thanks for the PR, one comment/ask for you.

File: charts/cluster-autoscaler/Chart.yaml (hunk 1 above)
Comment (pierluigilenoci) [reply charts/cluster-autoscaler/Chart.yaml]: Done

Comment (pierluigilenoci) [reply charts/cluster-autoscaler/Chart.yaml]: Done

Comment (pierluigilenoci) [reply charts/cluster-autoscaler/Chart.yaml]: Done

**Comment by gjtempleton** [comment]
Thanks
/lgtm
/approve

**Pull request merged**